src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -105,13 +105,25 @@ class CredentialType(Enum):  # pylint: disable=too-few-public-methods
 
 class Profile(object):
 
-    def __init__(self, storage=None, auth_ctx_factory=None, async_persist=True, cli_ctx=None):
+    _global_creds_cache = None
+
+    def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=True,
+                 async_persist=True, cli_ctx=None):
         from azure.cli.core import get_default_cli
 
         self.cli_ctx = cli_ctx or get_default_cli()
         self._storage = storage or ACCOUNT
         self.auth_ctx_factory = auth_ctx_factory or _AUTH_CTX_FACTORY
-        self._creds_cache = CredsCache(self.cli_ctx, self.auth_ctx_factory, async_persist=async_persist)
+
+        if use_global_creds_cache:
+            # for perf, use global cache
+            if not Profile._global_creds_cache:
+                Profile._global_creds_cache = CredsCache(self.cli_ctx, self.auth_ctx_factory,
+                                                         async_persist=async_persist)
+            self._creds_cache = Profile._global_creds_cache
+        else:
+            self._creds_cache = CredsCache(cli_ctx, self.auth_ctx_factory, async_persist=async_persist)
```

Comment:
`CredsCache(self.cli_ctx, ....)`

Reply:
Okay. I assume you like for consistency, as the current code still functions

Reply:
this wouldn't pick up the result of `get_default_cli()` if no cli_ctx was passed in

Reply:
Good find. I didn't realize that line was touched by knack as well. This branch so far only gets used by tests, so the error has not surfaced.